matcher.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package pubsub
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// Match implements TopicMatcher. One topic matches another if they
+// are equal.
+func (t Topic) Match(topic Topic) bool {
+	return t == topic
+}
+
+type regexMatcher struct {
```

Comment:
It might be worth exposing this, and the field, so users aren't forced into dealing with panics for dynamically generated regexps.